src/main/java/systems/crigges/jmpq3/JMpqEditor.java

```diff
@@ -120,6 +120,8 @@
 
     /** If write operations are supported on the archive. */
     private boolean canWrite;
+    /** If the archive was originally read-only */
+    private boolean openedAsReadOnly;
```

Comment:
why add this superfluous variable?

Reply:
It seems like the "edit mode" is only available in the Constructor. If the (listfile) doesn't exist, it sets canWrite to "false" even if it was originally not opened as read-only. That's why the setExternalListfile method needs to set canWrite to "true". However I don't want to set canWrite to "true" if the archive was orignally opened as readOnly. Does this make sense? Maybe there's a better way of accomplishing this.

Reply:
could pass listfile in constructor i suppose to prevent setting canWrite to false.